### DISCUSSION TITLE
What is the maximum number of queues created in RabbitMQ?

### DISCUSSION THREAD

**Discussion by chienhust** [rabbitmq/rabbitmq-server#8243] [Questions] [answered]
I can't find out any document mentions about limitation of number of queues.
I refer to 4.9 Limitations in this document [ref](https://www.rabbitmq.com/resources/specs/amqp0-9-1.pdf) but I don't also see it.
Can you tell me the maximum/highest number of queues created in RabbitMQ?
Thank you very much.

**Comment by mkuratczyk** [comment, marked as answer]
There is no enforced limit. There are systems with hundreds of thousands of queues per node, but it doesn't mean you can just assume you can do that. It highly depends on the workload and queue types - there is no chance to achieve this with replicated queues for example. Similarly, if there is a lot of traffic per queue, it'd be hard to get anywhere near such numbers (unless you have a machine with thousands of cores I guess).

If you share details about your workload: messages per second, message size, number of connections, how many queues do you think you'd need, etc, it will be much easier to tell how feasible this could be.
